File: QueryOptimizer.Shared/Common/Models/ExecutionPlan/ExecutionPlanNode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QueryOptimizer.Shared.Common.Models.ExecutionPlan
{
    public class ExecutionPlanNode
    {
        public string NodeType { get; set; } = default!;

        public string NormalizedNodeType { get; set; } = default!;

        public string? ObjectName { get; set; } = default!;

        public string? IndexName { get; set; } = default!;

        public string? Predicate { get; set; } = default!;

        public string? JoinType { get; set; } = default!;

        public double? EstimatedCost { get; set; }

        public long? EstimatedRows { get; set; }

        public long? ActualRows { get; set; }

        public double? ActualTimeMs { get; set; }

        public long? LogicalReads { get; set; }

        public long? PhysicalReads { get; set; }

        public List<ExecutionPlanNode> ChildrenNode { get; set; } = new();
    }
}

File: QueryOptimizer.API/app/query-optimizer-front/src/app/models/executionPlanNode.ts
export interface ExecutionPlanNode {
  nodeType?: string | null;

  normalizedNodeType?: string | null;

  objectName?: string | null;

  indexName?: string | null;

  predicate?: string | null;

  joinType?: string | null;

  estimatedCost?: number | null;

  estimatedRows?: number | null;

  actualRows?: number | null;

  actualTimeMs?: number | null;

  logicalReads?: number | null;

  physicalReads?: number | null;

  children?: ExecutionPlanNode[];
}